app/scripts/services/authorization.js

```diff
@@ -96,16 +96,22 @@ angular.module("openshiftConsole")
       return _.get(cachedRulesByProject.get(projectName || currentProject), ['rules']);
     };
 
+    var checkRulesForResourceVerb = function(rules, resource, verb) {
+      return _.contains(rules[resource], verb) || _.contains(rules[resource], '*');
+    };
+
     var canI = function(verb, resource, projectName) {
```

Comment:
why not just pass the data in a structured way? you can keep the string as a simple case (api group = ''), and pass structured data in other cases. make the first line of `canI` do this:
```
resource = APIService.toResourceGroupVersion(resource);
```
and wherever you want to pass a non-empty API group, do `canI('create',{group:'extensions',resource:'horizontalpodautoscalers'})`